cpp/include/raft/stats/mean_center.cuh

```diff
@@ -51,6 +52,44 @@ void meanCenter(Type* out,
   detail::meanCenter<Type, IdxType, TPB>(out, data, mu, D, N, rowMajor, bcastAlongRows, stream);
 }
 
+/**
+ * @brief Center the input matrix wrt its mean
+ * @tparam DataT the data type
+ * @tparam IdxType index type
+ * @tparam LayoutPolicy Layout type of the input matrix.
+ * @tparam TPB threads per block of the cuda kernel launched
+ * @param handle the raft handle
+ * @param data input matrix
+ * @param mu the mean vector
+ * @param out the output mean-centered matrix
+ * @param bcastAlongRows whether to broadcast vector along rows or columns
+ */
+template <typename DataT, typename IdxType, typename LayoutPolicy, int TPB = 256>
```

Comment:
We shouldn't be exposing any cuda-specific templates so we can remove TPB from the public API